R/repository.R

```diff
@@ -1,32 +1,90 @@
-#' Create or open a git repository
+#' Create or discover a local git repository
```

Comment:
I grouped the functions into help topics like so:

* User-facing functions, take a path, return a path: `git_init()`, `git_find()`.
* Developer or programmer-facing functions, so mentions of a pointer to a git repo are in scope.